conf/provisioning/users/sample.yaml

```diff
@@ -0,0 +1,9 @@
+# # config file version
+apiVersion: 1
+
+users:
+  - name: Fake
+    login: fake-user
+    email: example@example.com
+    password: "hello"
```

Comment:
I don't like encouraging clear text password provisioning. Particularly without a "change on first login" behavior. We might not have the methods required to do this in a nice way but at least we should _allow_ copying a value fetched from another Grafana instance here or maybe send out a reset password email when the user is first provisioned.

Think something like:

```yaml
password:
    source: "password-reset-email"
```

```yaml
password:
    source: "raw"
    value: "a4f4f1781236e4ad97de347d1ae29a1721b28fe010c2acd6120a4c3ce4c88b614e7060f613a16c1cc502e424a0be1c837f8d"
```

```yaml
password:
    source: "plaintext"
    value: "hello"
```

Reply:
I generally agree.

I think that is password should be at least defined under `secureJsonData.password` similarly to what happens for the [datasource provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/#example-datasource-config-file) so as it will be stored encoded in the database.

Reply:
> I don't like encouraging clear text password provisioning

In theory, I agree. In practice, I would say that if security is a concern this feature should not be used. Which is why Im hesitant about it. 

Password/tokens/apikeys for datasource provisioning are in cleartext. To increase the security in Grafana we should solve that first :)

> at least we should allow copying a value fetched from another Grafana instance 

This would require the same `secret_key`. Security-wise good but it feels clunky.

Reply:
> In theory, I agree. In practice, I would say that if security is a concern this feature should not be used. Which is why Im hesitant about it.

I agree with that. I'm a little reluctant to open up for that. Some users might be very well aware and handle the problem with security concerns appropriately, but we can also be rather certain that others won't. It's a balance where I tend to lean towards "first, make it secure".

> This would require the same secret_key. Security-wise good but it feels clunky.

Yeah, I started realizing that half-way through writing that which was why I added triggering the reset password mechanism as an alternative to that.

Reply:
My initial goal with this PR was to mirror the [Admin HTTP API for creating users](https://grafana.com/docs/grafana/latest/http_api/admin/#global-users)  which has cleartext passwords, but I agree that's not very practical for any production deployments.

Having plaintext passwords as an option at all seems like an easy way for someone to shoot themselves in the foot later on, should there be just 1 way to set passwords for provisioned users (`source: "raw"` above)? It looks a little clunky but it makes sense and people should only have to do it once. And just to confirm my understanding, the user would have to:

1. set the `secret_key` a grafana instance
2. manually create the user
3. copy the hashed password to the provisioning file and make sure the `secret_key` is the same

Reply:
@bergquist  @sakjur 
what do you think about accepting only [environmental variables](https://grafana.com/docs/grafana/latest/administration/provisioning/#using-environment-variables) as password values as workaround?